golog.go

```diff
@@ -139,27 +147,16 @@ type Logger interface {
 	// Fatalf logs to stderr and then exits with status 1
 	Fatalf(message string, args ...interface{})
 
-	// Trace logs to stderr only if TRACE=true
```

Comment:
What replaces trace logging?

Reply:
Oh, I see, in Zap the "debug" level is analogous to what we used for trace. If we adopt those semantics, we'd have to go in and change all of our "debug" to "info" and "trace" to "debug".

Reply:
You'll hit all the libraries we use too?

Reply:
Yeah exactly- just search and replace.

Reply:
Yeah exactly- just search and replace.

Reply:
Okay. To be safe, it would be nice create a new interface instead of changing the existing one. Have `LoggerFor()` return an interface that has trace and debug, and also create a new function like `NewStyleLogerFor()` that has debug and info. I'm thinking here mostly programs other than flashlight that also use golog, like zeno and a bunch of other stuff that I've written.

Reply:
I anticipate more required changes, like the `TRACE=true` flag we got used to. I'm also a bit worried about the migration may never fully complete and leave two sets of API to co-exist. How important it is to change the semantics?

Reply:
I suppose another option would be to break the interface completely, so that stuff that's using the old API fails to compile. The problem right now is that unless something is using the Trace function, it will actually continue to compile but the behavior of `log.Debug` will be different than before.

Reply:
Yeah so I just committed a change that just uses a global `golog.ZapLogger` function, and we can convert existing libraries as desired. That makes it still a simple search and replace for flashlight and dependencies, for example, along with the required INFO/TRACE/DEBUG fixes. Basically you get the existing behavior with TRACE=true if you set the logger debug level to DEBUG, which is also available via a configuration file.

This would only affect zeno if desired, but arguably something like zeno would benefit quite a bit from using zap's speed improvements and great configurability.

The big problem with expanding the interface is note getting the structured logging benefits -- this latest version just makes that all available to callers.

Reply:
Tempting to just make this a separate library -- gonna do that...